tools/windows/DatadogAgentInstaller/CustomActions/UserCustomActions.cs

```diff
@@ -47,79 +49,72 @@ private static string GetDefaultDomainPart()
+                        session.Log($"Found {propertyName} in registry {session[propertyName]}");
                     }
-                    catch { }
-                }
-                else
-                {
-                    session.Log($"User provided DDAGENTUSER_NAME {session["DDAGENTUSER_NAME"]}");
                 }
+                catch { }
+            }
+            else
+            {
+                session.Log($"User provided {propertyName} {session[propertyName]}");
+            }
+        }
 
-                // PROJECTLOCATION
-                //
-                if (string.IsNullOrEmpty(session["PROJECTLOCATION"]))
-                {
-                    try
-                    {
-                        session["PROJECTLOCATION"] = Microsoft.Win32.Registry.LocalMachine.OpenSubKey(@"Software\Datadog\Datadog Agent")
-                            .GetValue("InstallPath")
-                            .ToString();
-                        session.Log($"Found PROJECTLOCATION in registry {session["PROJECTLOCATION"]}");
-                    }
-                    catch {  }
-                }
-                else
+        private static void RegistryValueProperty(ISession session, string propertyName, Microsoft.Win32.RegistryKey registryKey, string registryValue)
+        {
+            // Convenience wrapper of RegistryProperty for properties that have an exact 1:1 mapping to a registry value
+            // and don't require additional processing.
+            RegistryProperty(session, propertyName,
+                RegistryPropertyHandler =>
                 {
-                    session.Log($"User provided PROJECTLOCATION {session["PROJECTLOCATION"]}");
-                }
+                    return registryKey.GetValue(registryValue).ToString();
+                });
+        }
 
-                // APPLICATIONDATADIRECTORY
-                //
-                if (string.IsNullOrEmpty(session["APPLICATIONDATADIRECTORY"]))
-                {
-                    try
-                    {
-                        session["APPLICATIONDATADIRECTORY"] = Microsoft.Win32.Registry.LocalMachine.OpenSubKey(@"Software\Datadog\Datadog Agent")
-                            .GetValue("ConfigRoot")
-                            .ToString();
-                        session.Log($"Found APPLICATIONDATADIRECTORY in registry {session["APPLICATIONDATADIRECTORY"]}");
-                    }
-                    catch {  }
-                }
-                else
-                {
-                    session.Log($"User provided APPLICATIONDATADIRECTORY {session["APPLICATIONDATADIRECTORY"]}");
+        private static ActionResult ReadRegistryProperties(ISession session)
+        {
+            // This CA runs (only once) in either the InstallUISequence or the InstallExecuteSequence.
+            try
+            {
+                using (var subkey = Microsoft.Win32.Registry.LocalMachine.OpenSubKey(@"Software\Datadog\Datadog Agent")) {
```

Comment:
I believe that subkey can be null and need to be checked ... of course technically it can be voided because "catch" would catch it.